toolkit/tools/pkg/imagecustomizerlib/imagecreator.go

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package imagecustomizerlib
+
+import (
+	"path/filepath"
+	"time"
+
+	"github.com/microsoft/azurelinux/toolkit/tools/imagecustomizerapi"
+	"github.com/microsoft/azurelinux/toolkit/tools/imagegen/diskutils"
+	"github.com/microsoft/azurelinux/toolkit/tools/internal/logger"
+	"github.com/microsoft/azurelinux/toolkit/tools/internal/safechroot"
+)
+
+func CustomizeImageHelperImageCreator(buildDir string, baseConfigPath string, config *imagecustomizerapi.Config,
+	rawImageFile string, rpmsSources []string, useBaseImageRpmRepos bool,
+	imageUuidStr string, packageSnapshotTime string, tarFile string,
+) (map[string]diskutils.FstabEntry, string, error) {
+	logger.Log.Debugf("Customizing OS with imager")
+
+	imageConnection, partUuidToFstabEntry, _, err := connectToExistingImage(rawImageFile, buildDir, imageRoot, true)
+	if err != nil {
+		return nil, "", err
+	}
+	defer imageConnection.Close()
+
+	toolsChrootDir := filepath.Join(buildDir, toolsRoot)
+
+	toolsChroot, err := safechroot.CreateToolsChroot(toolsChrootDir, false, nil, nil, true, tarFile)
+	if err != nil {
+		return nil, "", err
+	}
+	defer toolsChroot.Close(false)
+
+	// Do the actual customizations.
+	err = doOsCustomizationsImageCreator(buildDir, baseConfigPath, config, imageConnection, toolsChroot, rpmsSources,
+		useBaseImageRpmRepos, imageUuidStr,
+		partUuidToFstabEntry, packageSnapshotTime)
+	// Out of disk space errors can be difficult to diagnose.
+	// So, warn about any partitions with low free space.
+
+	warnOnLowFreeSpace(buildDir, imageConnection)
+	if err != nil {
+		return nil, "", err
+	}
+
+	// Close the tools chroot and image connection.
+	toolsChroot.Close(false)
+	err = imageConnection.CleanClose()
+	if err != nil {
+		return nil, "", err
+	}
+
+	return partUuidToFstabEntry, "", nil
+}
+
+func doOsCustomizationsImageCreator(
```

Comment:
You should also include running the `postCustomization` and `finalizeCustomization` scripts.